src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -1919,6 +1928,24 @@ static void top_toolbar_tool_update_land_paint(sint16 x, sint16 y){
 *  rct2: 0x00664280
 */
 static void top_toolbar_tool_update_land(sint16 x, sint16 y){
+
+    uint8 side;
+
+    if (!gMapCtrlPressed)
+    {
+        if (input_test_place_object_modifier(PLACE_OBJECT_MODIFIER_COPY_Z))
+        {
+            gMapCtrlPressed = true;
+        }
+    }
+    else
+    {
+        if (!input_test_place_object_modifier(PLACE_OBJECT_MODIFIER_COPY_Z))
+        {
+            gMapCtrlPressed = false;
+        }
+    }
```

Comment:
This entire block can be replaced with:
`gMapCtrlPressed = input_test_place_object_modifier(PLACE_OBJECT_MODIFIER_COPY_Z)`